install/kubernetes/helm/istio/files/injection-template.yaml

```diff
@@ -263,6 +263,15 @@ containers:
   - name: STACKDRIVER_TRACING_MAX_NUMBER_OF_MESSAGE_EVENTS
     value: "{{ .ProxyConfig.GetTracing.GetStackdriver.GetMaxNumberOfMessageEvents }}"
   {{- end }}
+  {{ $envs := .Values.global.proxy.env }}
+  {{- if (isset .ObjectMeta.Annotations `sidecar.istio.io/environmentOverride`) }}
```

Comment:
I am leaning towards this being too complex here. Are there use cases for setting this per pod, or do we only need the global level?

Reply:
This PR should allow users to define global values for proxy environment variable via `global.proxy.env` AND override those values pod per pod. 
That way we would give users the flexibility to define those values at cluster level OR at pod level. Pod level variables override the global ones.
In my personal use case this would be very helpful, because pods in our cluster are very different from each other.
On the other side, that's the reason why I opened the PR as WIP, to see what the thoughts were on this approach.

Reply:
I have n't followed this fully - sorry if this is a naive question, does this help in setting resource limits per service?

Reply:
not really, I'm afraid.. :)
but for that purpose `sidecar.istio.io/proxyCPU` and a `sidecar.istio.io/proxyMemory` should help!

Reply:
but actually `sidecar.istio.io/proxyCPU` and a `sidecar.istio.io/proxyMemory` are a good example of the use case I would like to support with `sidecar.istio.io/environmentOverride` @howardjohn